webext/src/dom/commands_mixin.js

```diff
@@ -35,19 +41,148 @@ export default MixinBase =>
           break;
         case "/url":
           url = utils.rebuildArgsToSingleArg(parts);
-          document.location.href = url;
+          window.location.href = url;
+          break;
+        case "/url_up":
+          this.urlUp();
+          break;
+        case "/url_root":
+          window.location.href = window.location.origin;
           break;
         case "/history_back":
           history.go(-1);
           break;
+        case "/history_forward":
+          history.go(1);
+          break;
+        case "/reload":
+          window.location.reload();
+          break;
         case "/window_stop":
           window.stop();
           break;
+        case "/find_next":
+          this.findNext(parts[1]);
+          break;
+        case "/find_previous":
+          window.find(parts[1], false, true, false, false, true, true);
+          break;
+        case "/get_link_hints":
+          this.getLinkHints(false);
+          break;
+        case "/get_clickable_hints":
+          this.getLinkHints(true);
+          break;
+        case "/focus_first_text_input":
+          this.focusFirstTextInput();
+          break;
+        case "/follow_link_labeled_next":
+          this._followLinkLabeledNext();
+          break;
+        case "/follow_link_labeled_previous":
+          this._followLinkLabeledPrevious();
+          break;
         default:
           this.log("Unknown command sent to tab", message);
       }
     }
 
+    focusFirstTextInput() {
+      VimiumNormal.focusInput(1);
+    }
+
+    //adapted vimium code
+    followLinkLabeledNext() {
+      var nextPatterns = "next,more,newer,>,›,→,»,≫,>>,weiter" || "";
+      var nextStrings = nextPatterns.split(",").filter(function(s) {
+        return s.trim().length;
+      });
+      return (
+        VimiumNormal.findAndFollowRel("next") ||
+        VimiumNormal.findAndFollowLink(nextStrings)
+      );
+    }
+
+    _followLinkLabeledNext() {
+      this.followLinkLabeledNext();
+    }
+
+    //adapted vimium code
+    followLinkLabeledPrevious() {
+      var previousPatterns =
+        "prev,previous,back,older,<,‹,←,«,≪,<<,zurück" || "";
+      var previousStrings = previousPatterns.split(",").filter(function(s) {
+        return s.trim().length;
+      });
+      return (
+        VimiumNormal.findAndFollowRel("prev") ||
+        VimiumNormal.findAndFollowLink(previousStrings)
+      );
+    }
+
+    _followLinkLabeledPrevious() {
+      this.followLinkLabeledPrevious();
+    }
+
+    urlUp() {
```

Comment:
Can we have a little comment about what this does?

Reply:
This goes from www.domain.com/topic/suptopic/ to www.domain.com/topic/

Reply:
Thanks